x-pack/plugins/cloud_security_posture/server/types.ts

```diff
@@ -33,7 +34,7 @@ export interface CspServerPluginSetupDeps {
   // required
   data: DataPluginSetup;
   taskManager: TaskManagerSetupContract;
-
+  security: SecurityPluginStart;
```

Comment:
It's makes sense to use `SecurityPluginSetup` but in this case, there is a comment that says to use `securityPluginStart`

```
export interface SecurityPluginSetup {
  /**
   * @deprecated Use `authc` methods from the `SecurityServiceStart` contract instead.
   */
  authc: { getCurrentUser: (request: KibanaRequest) => AuthenticatedUser | null };

```

Reply:
`setup` and `start` are different methods and have different contracts. overriding it seems wrong. 

the correct type for the `setup` phase `SecurityPluginSetup`, we can't decide it's `SecurityPluginStart` as we don't own it. 

also seems like other usages define it like this:

```typescript
security?: SecurityPluginSetup
```

can you clarify wdym by "in this case" ? i don't see a case where we override an external plugin types. if it fixes an issue for us, then we may have another issue?

Reply:
Hey, @orouz please see the comment that security left on `SecurityPluginSetup`:

```
export interface SecurityPluginSetup {
  /**
   * @deprecated Use `authc` methods from the `SecurityServiceStart` contract instead.
   */
  authc: { getCurrentUser: (request: KibanaRequest) => AuthenticatedUser | null };
  /**
```

the `altering` plugin, doing the same, checkout:
`x-pack/plugins/alerting/server/alerting_authorization_client_factory.ts 18`

Reply:
@CohenIdo 

i see the comment, i'm asking how it relates to your case. 

i don't understand why you changed ` CspServerPluginSetupDeps` to include a `start` contract. i understand it fixes your issue, but i think this is the wrong fix. 

alerting plugin also doesn't change the setup deps to be start deps:

https://github.com/elastic/kibana/blob/9431231502fd389f4603add48faa43fb78de47bd/x-pack/plugins/alerting/server/plugin.ts#L134-L135

the example you shared is for their `AlertingAuthorizationClientFactory`, which they own, so it makes sense to use the types they need.